core/src/test/scala/org/apache/spark/sql/IssueTestSuite.scala

```diff
@@ -254,6 +254,20 @@ class IssueTestSuite extends BaseTiSparkSuite {
     judge("select cast(count(1) as char(20)) from `tmp_empty_tbl`")
   }
 
+  test("index double scan with predicate") {
```

Comment:
i suggest we add these tests to expression/index

Reply:
we also need to test aggregations and duplicate index columns.

Reply:
DONE